File: app/src/main/java/com/app/cheffyuser/home/adapter/FoodNearByAdapter.java
package com.app.cheffyuser.home.adapter;


import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.app.cheffyuser.R;
import com.app.cheffyuser.home.activities.FoodDetails;
import com.app.cheffyuser.home.model.FoodNearByModel;
import com.bumptech.glide.Glide;

import java.util.List;

public class FoodNearByAdapter extends RecyclerView.Adapter<FoodNearByAdapter.Holder>{

    private List<FoodNearByModel> foodlist;
    private Context context;

    public FoodNearByAdapter(List<FoodNearByModel> foodlist, Context context) {
        this.foodlist = foodlist;
        this.context = context;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_food_item_for_horizontal,parent, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final Holder holder, final int position) {

        holder.foodname.setText(foodlist.get(position).getName());
        holder.foodrating.setText(""+foodlist.get(position).getRating()+" (200)");
        holder.foodtime.setText(""+foodlist.get(position).getDeliveryTime()+" min");
       // holder.fooddeliver.setText(foodlist.get(position).getFood_delivery_text());
       // holder.foodimage.setImageResource(foodlist.get(position).getFood_image());

        String url=foodlist.get(position).getImageURL();

        //Loading image from url into imageView
        Glide.with(context)
                .load(url)
                .placeholder(R.drawable.image_placeholder)
                .error(R.drawable.image_placeholder)
                .into(holder.foodimage);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent a = new Intent(context, FoodDetails.class);
                context.startActivity(a);
            }
        });
    }

    @Override
    public int getItemCount() {
        return foodlist.size();
    }

    class Holder extends RecyclerView.ViewHolder {
        ImageView foodimage;
        TextView foodname,foodrating,foodtime,fooddeliver;
        Holder(@NonNull View itemView) {
            super(itemView);
            foodname = itemView.findViewById(R.id.foodname);
            foodrating = itemView.findViewById(R.id.food_ratings);
            foodtime = itemView.findViewById(R.id.times);
            fooddeliver = itemView.findViewById(R.id.deliverytext);
            foodimage = itemView.findViewById(R.id.foodimage);
        }
    }

}

File: app/src/main/java/com/app/cheffyuser/home/adapter/FoodNearByAdapter.kt
package com.app.cheffyuser.home.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.app.cheffyuser.R
import com.app.cheffyuser.home.model.FoodNearByModel
import com.app.cheffyuser.utils.loadUrl

class FoodNearbyAdapter(
    private val context: Context,
    private val foodNearbyModels: MutableList<FoodNearByModel>?,
    private val clickListener: RecyclerItemClickListener
) : RecyclerView.Adapter<BaseViewHolder>() {
    private var isLoaderVisible = false
    private lateinit var myHolder: BaseViewHolder

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.single_fooditem_horizontal, parent, false)
        return ViewHolder(view, clickListener)

    }

    override fun onBindViewHolder(holder: BaseViewHolder, position: Int) {
        myHolder = holder
        holder.onBind(position)
    }


    override fun getItemCount(): Int {
        return foodNearbyModels?.size ?: 0
    }

    fun add(foodNearByModel: FoodNearByModel) {
        foodNearbyModels!!.add(foodNearByModel)
        notifyItemInserted(foodNearbyModels.size - 1)
    }

    private fun remove(foodNearByModel: FoodNearByModel) {
        val position = foodNearbyModels!!.indexOf(foodNearByModel)
        if (position > -1) {
            foodNearbyModels.removeAt(position)
            notifyItemRemoved(position)
        }
    }

    private fun getItem(position: Int): FoodNearByModel? {
        return if (position != RecyclerView.NO_POSITION)
            foodNearbyModels!![position]
        else
            null
    }

    internal inner class ViewHolder(
        itemView: View,
        private val clickListener: RecyclerItemClickListener
    ) :
        BaseViewHolder(itemView), View.OnClickListener {
        private var model: FoodNearByModel? = null

        //init views here
        private val foodimage = itemView.findViewById<ImageView>(R.id.foodimage)
        private val foodname = itemView.findViewById<TextView>(R.id.foodname)
        private val food_ratings = itemView.findViewById<TextView>(R.id.food_ratings)
        private val times = itemView.findViewById<TextView>(R.id.times)
        private val deliverytext = itemView.findViewById<TextView>(R.id.deliverytext)

        init {

            foodimage.setOnClickListener(this)
            foodname.setOnClickListener(this)
            food_ratings.setOnClickListener(this)
            deliverytext.setOnClickListener(this)
            times.setOnClickListener(this)

        }

        override fun onBind(position: Int) {
            super.onBind(position)
            this.model = foodNearbyModels!![position]

            // set whatever you want. for instance;
            foodimage.loadUrl(model?.imageURL)
            foodname.text = model?.name
            food_ratings.text = model?.rating.toString()

            times.text = "${model!!.deliveryTime!!.minus(5)}-${model?.deliveryTime} min"

        }

        override fun onClick(view: View) {
            this.clickListener.modelClick(model!!)
        }

        override fun clear() {}
    }

    fun refreshList() {
        notifyDataSetChanged()
    }

    fun clearList() {
        val size = foodNearbyModels!!.size
        foodNearbyModels.clear()
        notifyItemRangeRemoved(0, size)
    }
}
